Sources/MapboxNavigation/PreviewViewController.swift

```diff
@@ -0,0 +1,654 @@
+        super.traitCollectionDidChange(previousTraitCollection)
+        
+        if previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
+            setupBottomBannerContainerViewLayoutConstraints()
+        }
+    }
+    
+    open override func viewSafeAreaInsetsDidChange() {
+        super.viewSafeAreaInsetsDidChange()
+        
+        setupTopBannerContainerViewLayoutConstraints()
+    }
+    
+    // MARK: - UIViewController setting-up methods
+    
+    func setupNavigationView() -> NavigationView {
+        let frame = parent?.view.bounds ?? UIScreen.main.bounds
+        let navigationView = NavigationView(frame: frame)
+        navigationView.navigationMapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        navigationView.navigationMapView.delegate = self
+        // TODO: Move final destination annotation placement logic to `MapView` or `NavigationMapView`.
+        navigationView.navigationMapView.mapView.mapboxMap.onNext(event: .styleLoaded) { [weak self] _ in
+            guard let self = self else { return }
+            self.pointAnnotationManager = self.navigationView.navigationMapView.mapView.annotations.makePointAnnotationManager()
+            
+            if let finalDestinationAnnotation = self.finalDestinationAnnotation,
+               let pointAnnotationManager = self.pointAnnotationManager {
+                pointAnnotationManager.annotations = [finalDestinationAnnotation]
+                
+                self.finalDestinationAnnotation = nil
+            }
+        }
+        
+        return navigationView
+    }
+    
+    func setupBackButton() {
+        let backButton = BackButton(type: .system)
+        backButton.translatesAutoresizingMaskIntoConstraints = false
+        // TODO: Add localization.
+        backButton.setTitle("Back", for: .normal)
+        backButton.clipsToBounds = true
+        backButton.isHidden = true
+        backButton.addTarget(self, action: #selector(didPressBackButton), for: .touchUpInside)
+        backButton.setImage(.backImage, for: .normal)
+        backButton.imageView?.contentMode = .scaleAspectFit
+        backButton.imageEdgeInsets = UIEdgeInsets(top: 10,
+                                                  left: 0,
+                                                  bottom: 10,
+                                                  right: 15)
+        navigationView.addSubview(backButton)
+        
+        self.backButton = backButton
+    }
+    
+    func setupFloatingButtons() {
+        cameraModeFloatingButton = FloatingButton.rounded(imageEdgeInsets: UIEdgeInsets(floatLiteral: 12.0)) as CameraModeFloatingButton
+        cameraModeFloatingButton.delegate = self
+        
+        navigationView.floatingButtons = [
+            cameraModeFloatingButton
+        ]
+        
+#if DEBUG
+        let debugFloatingButton = FloatingButton.rounded(image: .debugImage,
```

Comment:
Functionality will be added in different PR.

Reply:
Is this is a feature we’re adding long-term, or just a placeholder for a different floating button we’ll be adding by default? If the latter, remember to remove the debug icon from the asset catalog.

Reply:
Planning on keeping it long-term, it'll be useful mostly for debugging purposes (showing viewport padding, changing runtime styling, logging etc).